lib/GLPI/Agent/Task/Inventory/Generic/Dmidecode/Memory.pm

```diff
@@ -36,7 +37,7 @@ sub doInventory {
     # If only one component is defined and we are under a vmsystem, we can update
     # component capacity to real found size. This permits to support memory size updates.
     my $vmsystem = $inventory->getHardware('VMSYSTEM');
-    if ($vmsystem && $vmsystem ne 'Physical') {
+    if ($vmsystem && ! grep ( /^$vmsystem$/, @physmemsystems)) {
```

Comment:
Your modification is too complex to just also ignore the VMware case in addition with the Physical one. Try the following with removing all the changes on line 29-30 & the definition of `@physmemsystems`.
```suggestion
    if ($vmsystem && $vmsystem !~ /^Physical|VMware$/) {
```
Your update finally remove the purpose of this check: permit to support memory size update. How would update memory on VMware and finally is this still detected ?

Reply:
Yes. I agree. This update removes the memory size update for Vmware virtual machines. As you can see in attached files given in my issue, dmidecode gives good values for VM memory settings.
There is no need to replace value given by dmidecode command with value retreived from /proc/meminfo. This value doesn't represent the real VM memory size setting. If VMware virtual machine memory size is greater than 4 GB, dmidecode command returns more than one memory slot. In that case, the memory size update check is not used and memory slot total size is correct.
In my modification, I use an array because I don't know if only these too vmsystem types are concerned by this check. I thought it was easier to add a vmsystem type in the array if needed.
If you think an "or" regex pattern is better, let's do it  😄. I think the right pattern is: ^(Physical|VMware)$
With the pattern ^Physical|VMware$, Physicalsdsd or sdsdVMware values match pattern.

Reply:
Yes, I think the "or" regex pattern is better here. If this becomes required for another type we will just have to add it to the regex and after all there're a limited number of possible case.
Also you're right about the right pattern.
Can you update your PR ?

Reply:
Hello,
My PR has been updated wih pattern regex check.